sources/commands/option_external.c

```diff
@@ -40,13 +40,13 @@ static int	run_executable(char *fullpath, t_param *param)
 
 /* test with empty path, found in first, found in last, not found,
  * found but not allowed, empty command */
-int	option_external(int choice, t_treenode *node, t_param *param,
-					t_execution_result *res)
+int	option_external(t_control control, t_treenode *node, t_param *param)
 {
 	char	*fullpath;
 
-	if (!choice)
+	if (*control.found || !control.choice)
 		return (0);
+	*control.found = 1;
 	if (param->debug_output_level & DBG_EXTERNAL_SEARCH_FOLDERS)
```

Comment:
it's only for debugging, right? what is usage of it? when do you want to change the value of it?

Reply:
No. `choice` is a condition calculated on caller side and checked on command's side. That's to compare command with its text. `found` is a variable that's used to mark that one command's name has been parsed, so we don't have to check other commands. Otherwise I'd had to have to many lines in the caller (command choice function).

Reply:
@ioannnikulin sorry, i meant this line :D

if (param->debug_output_level & DBG_EXTERNAL_SEARCH_FOLDERS)

Reply:
Yep, that's for debug output. Currently I set its value in param_init (options are in minishell.h), later I think we'll pass it as a command line argument.